adam/experiment/log_experiment.py

```diff
@@ -294,6 +382,42 @@ def learner_factory_from_params(
             ),
             functional_learner=FunctionalLearner(language_mode=language_mode),
         )
+    elif learner_type == "integrated-object-only":
+        object_learner_type = params.string(
+            "object_learner_type",
+            valid_options=["subset", "pbv", "pursuit"],
+            default="subset",
+        )
+        learner_params = params.namespace_or_empty("learner_params")
+        if not learner_params:
+            learner_params = Parameters.empty(namespace_prefix="learner_params")
+        if object_learner_type == "subset":
+            object_learner = SubsetObjectLearnerNew(  # type: ignore
+                ontology=GAILA_PHASE_2_ONTOLOGY,
+                beam_size=beam_size,
+                language_mode=language_mode,
+            )
+        elif object_learner_type == "pbv":
+            object_learner = ProposeButVerifyObjectLearner.from_params(  # type: ignore
+                learner_params
+            )
+        elif object_learner_type == "pursuit":
+            object_learner = PursuitObjectLearnerNew(  # type: ignore
+                learning_factor=learner_params.floating_point("learning_factor"),
+                graph_match_confirmation_threshold=learner_params.floating_point(
+                    "graph_match_confirmation_threshold"
+                ),
+                lexicon_entry_threshold=learner_params.floating_point(
+                    "lexicon_entry_threshold"
+                ),
+                rng=rng,
+                smoothing_parameter=learner_params.floating_point("smoothing_parameter"),
+                ontology=GAILA_PHASE_2_ONTOLOGY,
+                language_mode=language_mode,
+            )
+        else:
+            raise RuntimeError(f"Invalid Object Learner Type Selected: {learner_type}")
+        return lambda: IntegratedTemplateLearner(object_learner=object_learner)
```

Comment:
Will we ever call this lambda more than once per execution? I assume not, but if we do then this might cause problems since the object learner will be the same every time.

If we will call it more than once, it might make more sense to `def` the learner-making function in each branch and return the function here. (I'm not sure if Python lets you define functions with the same name in different branches, but I can't see why it wouldn't.)

Reply:
@spigo900 I'm unsure if this would ever get called more than once per execution but I was following the templates above for how this function is returning learners as I figured it was best not to change the return.

Reply:
@LichMaster98 Change the return type? It shouldn't require changing the type. Or do you mean changing something else about the return?

Slightly different but largely the same -- you could also `def object_learner_factory()` in each branch and that would let you keep the return expression similar to the template. It would just change `object_learner=object_learner` to `object_learner=object_learner factory()`.

Reply:
To clarify, when I say each branch, I mean each branch of `if object_learner_type == ...`, not of the outer if-statement.

Reply:
Oh! That makes sense, I'll make that change. Thanks